core/primitives/src/views.rs

```diff
@@ -355,6 +356,81 @@ pub struct NetworkInfoView {
     pub known_producers: Vec<KnownProducerView>,
 }
 
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub enum SyncStatusView {
+    /// Initial state. Not enough peers to do anything yet.
+    AwaitingPeers,
+    /// Not syncing / Done syncing.
+    NoSync,
+    /// Syncing using light-client headers to a recent epoch
+    // TODO #3488
+    // Bowen: why do we use epoch ordinal instead of epoch id?
+    EpochSync { epoch_ord: u64 },
+    /// Downloading block headers for fast sync.
+    HeaderSync {
+        start_height: BlockHeight,
+        current_height: BlockHeight,
+        highest_height: BlockHeight,
+    },
+    /// State sync, with different states of state sync for different shards.
+    StateSync(CryptoHash, HashMap<ShardId, ShardSyncStatusView>),
+    /// Sync state across all shards is done.
+    StateSyncDone,
+    /// Catch up on blocks.
+    BodySync { start_height: BlockHeight, current_height: BlockHeight, highest_height: BlockHeight },
+}
+
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct CatchupStatusView {
+    pub sync_block_hash: CryptoHash,
```

Comment:
add a comment - are these the ones that we're currently at or the ones that we're syncing to ?